modules/reactor/src/meshgenerators/ReactorMeshParams.C

```diff
@@ -0,0 +1,100 @@
+  params.addParam<boundary_id_type>("top_boundary_id",
+                                    "The boundary ID to set on top boundary of the extruded mesh");
+  params.addParam<boundary_id_type>(
+      "bottom_boundary_id", "The boundary ID to set on bottom boundary of the extruded mesh");
+  params.addParam<boundary_id_type>(
+      "radial_boundary_id",
+      "The boundary ID to set on the outer radial boundary of a CoreMeshGenerator object");
+  params.addParam<std::vector<Real>>(
+      "axial_regions", std::vector<Real>(1), "Length of each axial region");
+  params.addParam<std::vector<unsigned int>>(
+      "axial_mesh_intervals",
+      std::vector<unsigned int>(1),
+      "Number of elements in the Z direction for each axial region");
+  params.addClassDescription("This ReactorMeshParams object acts as storage for persistent "
+                             "information about the reactor geometry.");
+  return params;
+}
+
+ReactorMeshParams::ReactorMeshParams(const InputParameters & parameters)
+  : MeshGenerator(parameters),
+    _dim(getParam<MooseEnum>("dim")),
+    _geom(getParam<MooseEnum>("geom")),
+    _assembly_pitch(getParam<Real>("assembly_pitch")),
+    _axial_regions(getParam<std::vector<Real>>("axial_regions")),
+    _axial_mesh_intervals(getParam<std::vector<unsigned int>>("axial_mesh_intervals"))
+{
+  if (_axial_regions.size() != _axial_mesh_intervals.size())
+    mooseError("The number of axial regions is not consistent with the number of axial intervals.");
+
+  this->declareMeshProperty("mesh_dimensions", int(_dim));
+  this->declareMeshProperty("mesh_geometry", std::string(_geom));
+  this->declareMeshProperty("axial_boundaries", _axial_regions);
+  this->declareMeshProperty("axial_mesh_intervals", _axial_mesh_intervals);
+  this->declareMeshProperty("assembly_pitch", _assembly_pitch);
+
+  if (isParamValid("top_boundary_id"))
+  {
+    _top_boundary = getParam<boundary_id_type>("top_boundary_id");
+    this->declareMeshProperty("top_boundary_id", _top_boundary);
+  }
+  if (isParamValid("bottom_boundary_id"))
+  {
+    _bottom_boundary = getParam<boundary_id_type>("bottom_boundary_id");
+    this->declareMeshProperty("bottom_boundary_id", _bottom_boundary);
+  }
+  if (isParamValid("radial_boundary_id"))
+  {
+    _radial_boundary = getParam<boundary_id_type>("radial_boundary_id");
+    this->declareMeshProperty("radial_boundary_id", _radial_boundary);
+    if (isParamValid("top_boundary_id") && _radial_boundary == _top_boundary)
+      mooseError("top_boundary_id and radial_boundary_id must be unique values");
+    if (isParamValid("bottom_boundary_id") && _radial_boundary == _bottom_boundary)
+      mooseError("bottom_boundary_id and radial_boundary_id must be unique values");
+  }
+  if (isParamValid("top_boundary_id") && isParamValid("bottom_boundary_id") &&
+      (_bottom_boundary == _top_boundary))
+    mooseError("top_boundary_id and bottom_boundary_id must be unique values");
```

Comment:
These are very generic names for properties and are not modular. they could be easily conflicted by mixing with other MGs

Reply:
Would something like `reactor_top` `reactor_bottom` `reactor_radial` be better?

Reply:
the other params are pretty generic too.
@loganharbour think we should prefix all of them?
I dont envision a conflict coming up right away

Reply:
Whether or not a conflict comes up right away, I think we should get it right for now. IMO we should have a method on `ReactorApp` that gives fills with prefix. This prefix can then be changed in the future via an action or whatever in case a conflict is ever found.

Reply:
ok let's declare the prefix as a mesh property too, that way it wont be needed as a parameter in all mesh generators?

Reply:
No, then we have another name to deal with. On `ReactorApp`, we have:

```
std::string meshPropertyName(const std::string & name) const
{
  return _mesh_property_prefix + "_" + name;
}

const std::string _mesh_property_prefix = "SOME_DEFAULT";
```

Which in the future we can add an interface to change if we need to.

Reply:
To clarify, all mesh properties should be defined with this prefix?

Reply:
so these mesh properties are associated with a mesh right? They dont have a global scope, so they would not conflict between MGs (unless they are acting on the same mesh) in the end right?

Reply:
Correct, mesh properties are set locally on the MG's, the only global ones I can think of are the boundary sidesets names

Reply:
Ok I think we agree that there's something to do here to disambiguate. However seems to be that a new interface is needed to do it cleanly so this goes out of scope for this PR. @loganharbour if you could please create an issue detailing what we'd need. I m somewhat clear on how the prefix would be added, I m not sure how the property would be retrieved unless you knew the prefix for whoever created it